### ISSUE TITLE
Implement a switch in the template for including/excluding browserstack

### ISSUE THREAD

**Issue by dannyprok** [hmrc/ui-test-template.g8#3]
As not all teams use browserstack so it would be useful to have the option to exclude it when creating a new repo.  

I.e.  creating a project without browserstack would look something like this:
```
name [ui-test-template]: my-tests
frontendServicePort [9949]:
productionRoute [auth-login-stub]: 
browserstack [no]:
```

And would result in the omission of the following lines/files/resources:
- uk.gov.hmrc.integration.cucumber.utils.RunnerBrowserStackTests
- uk.gov.hmrc.integration.cucumber.utils.driver.browsers.BrowserStack
- https://github.com/hmrc/ui-test-template.g8/blob/master/src/main/g8/src/test/scala/uk/gov/hmrc/integration/cucumber/utils/driver/Driver.scala#L14
- ./src/test/resources/browserstack
- run_browserstack.sh
- run_browserstachbinary.sh
- https://github.com/hmrc/ui-test-template.g8/blob/master/src/main/g8/build.sbt#L23

**Comment by CEPoole** [comment]
This can be fully implemented using Giter8s **truthy** mechanics apart from one line:
https://github.com/hmrc/ui-test-template.g8/blob/master/src/main/g8/src/test/scala/uk/gov/hmrc/test/ui/driver/Driver.scala#L29
This is because Giter8 only handles the creation of new files rather adding lines to existing pages.

There are four obvious approaches to fixing this:
-  Put the burden on the user to add the line themselves to gain access to BrowserStack
-  Leave a commented line in the code, regardless of the choice to use BrowserStack (:-1: )
-  A post-setup but pre-use shell script (similar to this migrate script in the frontend templates https://github.com/hmrc/hmrc-frontend-scaffold.g8/blob/master/src/main/g8/migrate.sh)
-  Have two _almost_ identical files that only differ in this one line and use **truthy** to select between them

This is based on my understanding of Giter8 but I'd been keen to hear if there is a different approach to this problem

**Comment by TomDudley92** [comment]
I like the idea of either having the post-setup script or the two _almost_ identical files. These ways seem the most user friendly and the cleanest I'd say

**Comment by CEPoole** [comment]
Just realised I shared half of the required migrate script, here is the part that does the heavy lifting in the mentioned example (ours would be **much** simpler hopefully)
https://github.com/hmrc/hmrc-frontend-scaffold.g8/blob/master/src/main/scaffolds/yesNoPage/migrations/%24className__snake%24.sh

**Comment by dannyprok** [comment]
@CEPoole @TomDudley92 
I'm leaning towards your first option:
- put the burden on the user to add the line themselves to gain access to BrowserStack

I'm not convinced that we want to support updating your ui-test project because it would mean laying out some quite strict coding standards for the updates to stick.  In this case, someone might decide to move the Driver class, or rewrite it in a way that doesn't lend itself to being updated.  Preventing teams from making these changes puts a restriction on them, and, will be difficult to police.  

If this is boilerplate code that doesn't change across project, then I think it's best implemented in a shared library and not the template.

**Comment by TomDudley92** [comment]
I'm happy for the end user to implement the browserstack functionality themselves as there are quiet a few examples on the platform for them to follow.

I don't know if I like it being moved into a shared library though as that will then be another dependency for the project and will potentially require a lot of maintenance to keep working (depends how much of the BrowserStack code is moved across to the library though). Unless again it was an open project like the ui-test-template that anyone could raise an issue with and fix then that would be alright

**Comment by dannyprok** [comment]
I've followed @CEPoole advice and implemented this functionality using scaffolds.  see pull request: https://github.com/hmrc/ui-test-template.g8/pull/14

As projects that apply the new browserstack scaffold may run their tests with Cucumber, I've included a cucumber truthy parameted in the browserstack scaffold.  Setting this to "true" will:
- include the Browserstack cucumber runner
- include an example Browserstack .feature file
- include the cucumber sbt command in the `run_browserstack.sh` script

Regarding the inclusion of the `browserstack` line in `Driver.scala`, I decided to go with Option 1, which was to "put the burden on the user...." and update the README.md with a step by step explaining how to get browserstack running (which is largely done). 

The only other option I liked was to include a script which writes the line into the file however I'm leaning on the side of simplicity.  Keen to hear other's thoughts.
